src/poetry/packages/locker.py

```diff
@@ -268,6 +268,7 @@ def __walk_dependencies(
             extras = requirement.extras
             requirement = locked_package.to_dependency()
             requirement.marker = requirement.marker.intersect(marker)
+            requirement._extras = extras
```

Comment:
```suggestion
            requirement._extras = extras
            requirement._features = frozenset(extras)
```

In deference to the change proposed in https://github.com/python-poetry/poetry-core/pull/373/files#r882064718 this should allow us to update to next core without issues.

Reply:
The intention is that this MR fixes things with the current poetry-core, and when poetry-core exposes a better API we use it here - which I would expect to do with a later MR.

If you want to block this behind publishing poetry-core improvements I guess that's fine by me, then we can do it in a single commit here.

Reply:
I am simply anticipating a forward port required to get the downstream tests passing. And suggesting that we remove `requirement._extras = extras` once the new API is available.

I'd rather not block this behind a new core release and have this fixed in the next poetry pre-release that will likely happen prior to next core release.